build.gradle

```diff
@@ -17,140 +17,56 @@
-        maven { url = repos.spine }
-        maven { url = repos.spineSnapshots }
-    }
-
-    //noinspection GroovyAssignabilityCheck
-    dependencies {
-        classpath deps.build.guava
-        classpath(deps.build.gradlePlugins.protobuf) {
-            exclude group: 'com.google.guava'
-        }
-        classpath(deps.build.gradlePlugins.errorProne) {
-            exclude group: 'com.google.guava'
-        }
-        classpath "io.spine.tools:spine-model-compiler:$deps.versions.spine.base"
-    }
-}
-
-allprojects {
-    apply plugin: 'maven'
-    apply plugin: 'jacoco'
-    apply plugin: 'idea'
+plugins {
+    id 'io.spine.tools.gradle.bootstrap' version '1.0.0-SNAPSHOT' apply false
+    id 'net.ltgt.errorprone' version '0.6' apply false
 }
 
 subprojects {
-
-    repositories {
-        jcenter()
-        google()
-        mavenCentral()
-        mavenLocal()
-        maven { url = repos.gradlePlugins }
-        maven { url = repos.spine }
-        maven { url = repos.spineSnapshots }
-    }
-
-    configurations {
-        // Avoid collisions of Java classes defined both in `protobuf-lite` and `protobuf-java`
-        runtime.exclude group: "com.google.protobuf", module: "protobuf-lite"
-        testRuntime.exclude group: "com.google.protobuf", module: "protobuf-lite"
-    }
-
     project.ext {
-        spineProtobufPluginId = 'io.spine.tools.spine-model-compiler'
-
         sourcesRootDir = "$projectDir/src"
         generatedRootDir = "$projectDir/generated"
+    }
 
-        generatedJavaDir = "$generatedRootDir/main/java"
-        generatedTestJavaDir = "$generatedRootDir/test/java"
+    apply plugin: 'io.spine.tools.gradle.bootstrap'
 
-        generatedGrpcDir = "$generatedRootDir/main/grpc"
-        generatedTestGrpcDir = "$generatedRootDir/test/grpc"
+    spine {
+        project.defaultRepositories(project)
 
-        generatedSpineDir = "$generatedRootDir/main/spine"
-        generatedTestSpineDir = "$generatedRootDir/test/spine"
+        java()
     }
 
-    apply plugin: 'java'
-    apply plugin: 'com.google.protobuf'
     apply plugin: 'net.ltgt.errorprone'
-    apply plugin: 'pmd'
-
-    apply from: "$rootDir/pmd/pmd.gradle"
-
-    tasks.withType(JavaCompile) {
-
-        // Configure Error Prone:
-        // 1. Exclude generated sources from being analyzed by Error Prone.
-        // 2. Turn the check off until Error Prone can handle `@Nested` JUnit classes.
-        //    See issue: https://github.com/google/error-prone/issues/956
-        //
-        // For more config details see: 
-        //    https://github.com/tbroyer/gradle-errorprone-plugin/tree/v0.6#usage
-
-        options.errorprone.errorproneArgs << "-XepExcludedPaths:.*/generated/.*" << "-Xep:ClassCanBeStatic:OFF"
-    }
+    apply from: "$rootDir/script/pmd/pmd.gradle"
+    apply from: "$rootDir/script/idea.gradle"
+    apply from: "$rootDir/script/tests.gradle"
 
     sourceCompatibility = 1.8
     targetCompatibility = 1.8
 
+    configurations {
```

Comment:
This is something we would want to configure with the plugin. It's really a magical piece to those who never had a word with Spine. Also, Spine does not seem to work properly, if these aren't excluded.

Reply:
Please see [this issue](https://github.com/SpineEventEngine/bootstrap/issues/6) in `bootstrap` for the further improvements.

We will eliminate the need to configure these things, in an iterative manner.